src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const { id, title, completed, userId } = selectedTodo;
```

Comment:
You are destructuring `selectedTodo` to get `userId` which is a prop. This is not recommended because it can lead to bugs if the parent component changes the prop value. It's better to use the prop directly where it's needed.